glusterd2/peer/self.go

```diff
@@ -39,18 +39,25 @@ func normalizeAddrs() ([]string, error) {
 
 // AddSelfDetails results in the peer adding its own details into etcd
 func AddSelfDetails() error {
-	var err error
 
+	var err error
 	p := &Peer{
 		ID:            gdctx.MyUUID,
 		Name:          gdctx.HostName,
 		PeerAddresses: []string{config.GetString("peeraddress")},
 	}
-
 	p.ClientAddresses, err = normalizeAddrs()
 	if err != nil {
 		return err
 	}
+	peerInfo, err := GetPeer(gdctx.MyUUID.String())
+	if err != nil {
+
+		p.Metadata = make(map[string]string)
```

Comment:
You have check `err` and handle `errors.ErrPeerNotFound` separate from rest of the errors. Peer not found is not an error. Can't access the store is an error.